### ISSUE TITLE
Recognizer.removeEventListener 

### ISSUE THREAD

**Issue by stevenlimcorn** [ccoreilly/vosk-browser#32]
I am currently using Vue Js to run Vosk-browser and manage to call the ASR model and Kaldi recognizer by using 
```
this.recognizer.on("result", (message) => {
    const result = message.result;
    this.full.textContent += result.text + " "
})
```
The model is working well, however, I am trying to remove the event listener by using:
```
this.recognizer.removeEventListener("result", (message) => {
    const result = message.result;
    this.full.textContent += result.text + " "
})
```
Is this the way of doing it?

**Comment by ccoreilly** [comment]
Hello @stevenlimcorn !

good catch! Adding an EventListener to a recognizer instance will actually add it to the model class (which handles communication with the worker)

You can achieve what you want by removing the event listener from the model instead.

I'll extend the api so it works as one would expect. Thanks for pointing it out.

**Comment by xadh00m** [comment]
Hi @ccoreilly!

Thank you for this great wrapper library. If I understand the code correctly the listener added to the model is actually an anonymous wrapper function. Therefore it seems to be impossible to unregister it cleanly at the moment.

https://github.com/ccoreilly/vosk-browser/blob/c7877a6f2a0dc4f9b2bcb2a9138276cb8f0dbdeb/lib/src/model.ts#L74

**Comment by xadh00m** [comment]
I´m wondering why the KaldiRecognizer class extends EventTarget but uses the "on" method approach for registration. I could provide a PR to fix the issue but am unsure which way to go here.

**Comment by ccoreilly** [comment]
Hi @xadh00m ! Thank you very much for looking into it.

It was a mistake to register recognizer event listeners to the Model class and as you noted the goal was to make each KaldiRecognizer an EventTarget but somehow got overseen. The reason for adding the `on` method was to offer a simple API with type hints, but I realize it makes it impossible to unregister event listeners.

I have made some changes in PR #38 that will allow adding and removing event listeners to recognizer instances as long as you use the `addEventListener` and `removeEventListener`. However, if using that way you will have to check for the message inside `event.detail` but I guess that is expected. I would not like to change the `on` method as it will break code already using it.

What do you think? (cc @stevenlimcorn )

**Comment by xadh00m** [comment]
Hi @ccoreilly,

Ah ok I see now that you want to keep the "on" method for backwards compatibility. Looks good to me.

Thank you very much! As soon as you land the PR I will test this approach.

**Comment by ccoreilly** [comment]
Fixed in v0.0.7

**Comment by xadh00m** [comment]
Hi @ccoreilly 

Great news. Thank you very much! I´m going to test it now. :-)
